kasa/tests/test_feature.py

```diff
@@ -181,7 +181,8 @@ async def _test_feature(feat, query_mock):
 
     async def _test_features(dev):
         exceptions = []
-        for feat in dev.features.values():
+        feats = dev.features.copy()
```

Comment:
renegotiation causes clearing the feature dict, so we need to make a copy here & mock the request_reneg below.

Reply:
Depending on whether we stick with `request_renegotiation` or not I think there needs to be more of a comment here explaining that one/some of the features will actually clear all the internal state when `set` so this code is handling it etc.